quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -940,82 +920,78 @@ impl Metastore for PostgresqlMetastore {
             "#,
-            )
-            .bind(delete_opstamp as i64)
-            .bind(index_id)
-            .bind(split_ids)
-            .execute(&mut *tx)
-            .await?;
+        )
+        .bind(delete_opstamp as i64)
+        .bind(index_id)
+        .bind(split_ids)
+        .execute(&self.connection_pool)
+        .await?;
 
-            // If no splits is affected, maybe the index itself does not exist
-            // in the first place.
-            if sqlx_result.rows_affected() == 0 && index_opt(tx, index_id).await?.is_none() {
-                return Err(MetastoreError::IndexDoesNotExist {
-                    index_id: index_id.to_string(),
-                });
-            }
-            Ok(())
-        })?;
+        // If no splits were updated, maybe the index does not exist in the first place?
+        if update_res.rows_affected() == 0
+            && index_opt(&self.connection_pool, index_id).await?.is_none()
+        {
+            return Err(MetastoreError::IndexDoesNotExist {
+                index_id: index_id.to_string(),
+            });
+        }
         self.update_index_update_timestamp(index_id).await;
         Ok(())
     }
 
-    /// Lists delete tasks with opstamp > `opstamp_start`.
-    #[instrument(skip(self))]
+    /// Lists the delete tasks with opstamp > `opstamp_start`.
+    #[instrument(skip(self), fields(index_id=index_id))]
     async fn list_delete_tasks(
         &self,
         index_id: &str,
         opstamp_start: u64,
     ) -> MetastoreResult<Vec<DeleteTask>> {
-        run_with_tx!(self.connection_pool, tx, {
-            let delete_tasks: Vec<postgresql_model::DeleteTask> =
-                sqlx::query_as::<_, postgresql_model::DeleteTask>(
-                    r#"
+        let pg_delete_tasks: Vec<postgresql_model::DeleteTask> =
+            sqlx::query_as::<_, postgresql_model::DeleteTask>(
+                r#"
                 SELECT * FROM delete_tasks
                 WHERE
                     index_id = $1
                     AND opstamp > $2
                 "#,
-                )
-                .bind(index_id)
-                .bind(opstamp_start as i64)
-                .fetch_all(tx)
-                .await?;
-            delete_tasks
-                .into_iter()
-                .map(|delete_task| delete_task.try_into())
-                .collect::<MetastoreResult<_>>()
-        })
+            )
+            .bind(index_id)
+            .bind(opstamp_start as i64)
+            .fetch_all(&self.connection_pool)
+            .await?;
+
+        pg_delete_tasks
+            .into_iter()
+            .map(|pg_delete_task| pg_delete_task.try_into())
+            .collect::<MetastoreResult<_>>()
     }
 
     /// Returns `num_splits` published splits with `split.delete_opstamp` < `delete_opstamp`
-    /// ordered by ASC `split.delete_opstamp`.
-    #[instrument(skip(self))]
+    /// ordered by ascending `split.delete_opstamp` and `split.publish_timestamp`.
```

Comment:
Nitpick: I don't think the wording reads great here, it reads like it's part of the first sentence but probably should be read as a separate sentence, I think something along these lines probably makes it a bit better, but it's not the end of the world.

```suggestion
    /// Returns `num_splits` published splits with `split.delete_opstamp` < `delete_opstamp`.
    ///
    /// Results are ordered by ascending `split.delete_opstamp` and `split.publish_timestamp` values.
```

or

```suggestion
    /// in ascending order of `split.delete_opstamp` and `split.publish_timestamp`.
```